src/app/Http/Controllers/Operations/InlineCreateOperation.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Backpack\CRUD\app\Http\Controllers\Operations;
+
+use Backpack\CRUD\app\Http\Controllers\CrudController;
+use Illuminate\Support\Facades\Route;
+
+trait InlineCreateOperation
+{
+    /**
+     * Define which routes are needed for this operation.
+     *
+     * @param string $segment    Name of the current entity (singular). Used as first URL segment.
+     * @param string $routeName  Prefix of the route name.
+     * @param string $controller Name of the current CrudController.
+     */
+    protected function setupInlineCreateOperationRoutes($segment, $routeName, $controller)
+    {
+        Route::get($segment.'/inline/create', [
+            'as'        => $segment.'-inline-create',
+            'uses'      => $controller.'@getInlineCreateModal',
+            'operation' => 'InlineCreateOperation',
+        ]);
+        Route::post($segment.'/inline/create', [
+            'as'        => $segment.'-inline-create-save',
+            'uses'      => $controller.'@storeInlineCreate',
+            'operation' => 'InlineCreateOperation',
+        ]);
+        Route::get($segment.'/inline/refresh', [
+            'as'        => $segment.'-inline-refresh-options',
+            'uses'      => $controller.'@inlineRefreshOptions',
+            'operation' => 'InlineCreateOperation',
+        ]);
+    }
+
+    public function setupInlineCreateDefaults()
+    {
+        $this->crud->setOperationSetting('inline_create', true);
+    }
+
+    public function getInlineCreateModal()
+    {
+        if (request()->has('entity')) {
+            $this->setupOperationSettings();
+
+            return $this->getModalContent(request()->get('entity'), 'create', $this->crud->getCreateFields());
+        }
+    }
+
+    public function setupOperationSettings()
+    {
+        if (method_exists($this, 'setupCreateOperation')) {
+            $this->setupCreateOperation();
+        } else {
+            $this->setup();
```

Comment:
```php
  /**
     * Setup operation default settings. We run setup() and setupCreateOperation() because those are run in middleware
     * and to get the fields we need them earlier in application lifecycle
     */
    protected function setupInlineCreateDefaults()
    {
        $this->crud->setOperationSetting('inline_create', true);
        if (method_exists($this, 'setupCreateOperation')) {
            if (method_exists($this, 'setup')) {
                $this->setup();
            }
            $this->setupCreateOperation();
        } else {
            $this->setup();
        }
        $this->crud->applyConfigurationFromSettings('create');
    }
```

Reply:
Ah ok I understand now 😄